### ISSUE TITLE
server process (PID nnnn) exited with exit code 1

### ISSUE THREAD

**Issue by radcapitalist** [brianc/node-postgres#2535]
node.js version:  14.15.4
node-postgres version: 8.5.1
sequelize version: 5.12.13
pg-promise version: 10.8.7
OS: Windows

Hey, 

I using the sequelize + node-postgres stack, and also pg-promise (for large object interactions).  I have a situation where, once nightly at varying times, one of the postgres connection processes appears to be killed (first line below):

2021-04-10 03:11:46 EDT LOG:  server process (PID 6508) exited with exit code 1
2021-04-10 03:11:46 EDT DETAIL:  Failed process was running: COMMIT;
2021-04-10 03:11:46 EDT LOG:  terminating any other active server processes
2021-04-10 03:11:46 EDT WARNING:  terminating connection because of crash of another server process
2021-04-10 03:11:46 EDT DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2021-04-10 03:11:46 EDT HINT:  In a moment you should be able to reconnect to the database and repeat your command.

The "kill switch" in PostgreSQL subsequently takes down all connections, PostgreSQL goes into recovery mode for maybe 20 seconds or so, my application spins down and restarts, and ultimately all is well.  But it is disconcerting that this is happening.

The **exit code 1** bit has caused folks on the pgsql-general list to suggest that something external is taking down the PostgreSQL process.  I don't think it is **pg_terminate_backend()**, because the signature in the Postgres log is different when that happens.  I can match the log signature exactly by killing one of the postgres processes in Task Manager.

I'm frankly not sure who would have the pid to kill a PostgreSQL connection other than the code in this stack.  But I've hunted around in the source code and not found much in the way of **process.kill()**.  

Any thoughts on whether anything in this stack could be responsible for killing the postgres connection process and setting off this chain of events?

Thanks!

Eric

**Comment by sehrope** [comment]
Unless you're explicitly issuing kill commands as the PG super user, there's no way for this library to crash the server.

If any process crashes, the PG server will immediately terminate every other process to prevent further corruption. Try logging your server process IDs to see where the errant process is originating. That log snippet says it was processing a COMMIT so maybe a hardware failure during fsync?

**Comment by radcapitalist** [comment]
Thanks, @sehrope.  I do have the pids in the psql log.  The statement being processed by the process that exits tends to vary quite a bit.  I just haven't found any patterns in the time the process exits and other events that I can find going on in the system.  I added some Sequelize logging, since they helpfully use the debug module, and I see connections being destroyed at the time this is going on, but that may be more effect than cause.  Postgres log was not including milliseconds in timeouts for precise "which came first" analysis, until I changed it today.

So, thanks again, I'll keep hunting around.

Eric
